processing/src/main/java/io/druid/segment/data/GenericIndexed.java

```diff
@@ -449,64 +332,19 @@ public void writeToChannel(WritableByteChannel channel) throws IOException
    */
   public GenericIndexed<T>.BufferIndexed singleThreaded()
   {
-    if (valueBuffers.size() == 1) {
-      final ByteBuffer copyBuffer = valueBuffers.get(0).asReadOnlyBuffer();
-      return new BufferIndexed()
-      {
-        @Override
-        public T get(final int index)
-        {
-          checkIndex(index, size);
-
-          final int startOffset;
-          final int endOffset;
-
-          if (index == 0) {
-            startOffset = 4;
-            endOffset = headerBuffer.getInt(0);
-          } else {
-            int headerPosition = (index - 1) * Ints.BYTES;
-            startOffset = headerBuffer.getInt(headerPosition) + 4;
-            endOffset = headerBuffer.getInt(headerPosition + 4);
-          }
-          return _get(copyBuffer, startOffset, endOffset);
-        }
-      };
-    } else {
-
-      final List<ByteBuffer> copyValueBuffers = new ArrayList<>();
-      for (ByteBuffer buffer : valueBuffers) {
-        copyValueBuffers.add(buffer.asReadOnlyBuffer());
-      }
+    return versionOne ? singleThreadedVersionOne() : singleThreadedVersionTwo();
```

Comment:
benchmark of version 1 get call for this branching, just to make sure current version 1 has 0 effect because of this branching.